code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm

```diff
@@ -73,6 +116,27 @@
 	invocation_type = "none"
 	invocation = null
 
+/obj/effect/proc_holder/spell/aoe/conjure/build/darklight
+	name = "Summon Darklight"
+	desc = "This spell conjures a fragile crystal constructed of pure darkness. Acts as a negative light source."
+	action_icon_state = "shadow_darklight"
+	action_background_icon_state = "shadow_demon_bg"
+	school = "conjuration"
+	base_cooldown = 15 SECONDS
+	clothes_req = FALSE
+	invocation = "none"
+	invocation_type = "none"
+	summon_type = list(/obj/structure/darklight)
+
+/obj/effect/proc_holder/spell/aoe/conjure/build/darklight/can_cast(mob/living/simple_animal/demon/shadow/user, charge_check = TRUE, show_message = FALSE)
+	if(user.darklight_charges <= 0) // should never go less than 0 but let's just be safe
```

Comment:
you may just want to do darklight_charges < 1 here if byond does something stupid